src/components/screens/Authenticator/Authenticator.container.vue

```diff
@@ -0,0 +1,143 @@
+<template>
+  <auth-form-container
+    v-bind="$attrs"
+    :is-returnable="isReturnable"
+    v-on="$listeners"
+    @return="onReturn"
+  >
+    <template v-slot="{ isClosable }">
+      <router-view
+        v-bind="$attrs"
+        :is-closable="isClosable"
+        :is-returnable="isReturnable"
+        v-on="$listeners"
+        @complete="onComplete"
+        @switch="onSwitch"
+        @return="onReturn"
+      />
+    </template>
+  </auth-form-container>
+</template>
+
+<script>
+import AuthFormContainer from './Authenticator.interactor';
+
+export default {
+  name: 'AuthenticatorContainer',
+
+  props: {
+    isPasswordExist: {
+      type: Boolean,
+      default: false,
+    },
+
+    isOtp: {
+      type: Boolean,
+      default: false,
+    },
+  },
+
+  computed: {
+    isReturnable() {
+      return this.$route.name !== 'SignIn' && !!window.history.length;
+    },
+  },
+
+  methods: {
+    onReturn() {
+      if (!this.isReturnable) return;
+
+      this.$router.back();
+    },
+
+    onSwitch({ type }) {
+      switch (true) {
+        case type === 'sign-up':
```

Comment:
maybe define prefix for more readable, just like `to-sign-up`, and you see string and understand, that it will goto to next screen/container/page/etc. 
and searching in code will be easy with more unique strings
`sign-up` - is too abstact type i think

Reply:
I think we can just rename `type` to `to` (for example) without any prefixes.

Reply:
We should not bind on values, imo